src/Plugin/Field/FieldFormatter/MultiMediaFormatter.php

```diff
@@ -203,89 +197,23 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
-    if ($op == "responsive_image_style") {
-      $formatter = new MediaResponsiveImageFormatter($this->pluginId, $this->pluginDefinition, $this->fieldDefinition, $settings, $this->label, $view_mode, $this->thirdPartySettings, $this->loggerFactory, $this->entityTypeManager, $this->entityDisplayRepository);
-    }
+    $element['#stanford_media_image_style'] = ($op == "image_style") ? $this->settings['image']['image_formatter_image_style'] : $this->settings['image']['image_formatter_responsive_image_style'];
 
-    // Found one.
-    if ($formatter) {
-      return $formatter->viewElements($items, $langcode);
+    if ($op == 'image_style') {
+      $element['#pre_render'][] = [MediaImageFormatter::class, 'preRender'];
+      return;
     }
 
-    // Entity Render View Mode.
-    $view_mode = $this->settings['image']['image_formatter_view_mode'] ?: $view_mode;
-    return [$this->entityTypeManager->getViewBuilder('media')->view($media, $view_mode)];
-  }
-
-  /**
-   * Use view modes to generate a render array.
-   *
-   * @param \Drupal\Core\Field\FieldItemListInterface $items
-   *   A list of EntityReferenceItems.
-   * @param \Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem $item
-   *   A single item from the FieldItemList object.
-   * @param \Drupal\media\Entity\Media $media
-   *   An instantiated Media Object.
-   * @param string $langcode
-   *   A langcode string key. eg: en.
-   * @param string $view_mode
-   *   The view mode machine name.
-   *
-   * @return array
-   *   A render array for a view mode.
-   *
-   * @codeCoverageIgnore
-   *   Ignore this because in order to test it you have to mock everything up to
-   *   and including guzzle. A functional test has already been written for
-   *   oembed and can be extended. The functional test covers this function.
-   */
-  private function viewVideoElement(FieldItemListInterface $items, EntityReferenceItem $item, Media $media, $langcode, $view_mode = "default") {
-    return [$this->entityTypeManager->getViewBuilder('media')->view($media, $view_mode)];
-  }
-
-  /**
-   * Use view modes to generate a render array.
-   *
-   * @param \Drupal\Core\Field\FieldItemListInterface $items
-   *   A list of EntityReferenceItems.
-   * @param \Drupal\Core\Field\Plugin\Field\FieldType\EntityReferenceItem $item
-   *   A single item from the FieldItemList object.
-   * @param \Drupal\media\Entity\Media $media
-   *   An instantiated Media Object.
-   * @param string $langcode
-   *   A langcode string key. eg: en.
-   * @param string $view_mode
-   *   The view mode machine name.
-   *
-   * @return array
-   *   A render array for a view mode.
-   */
-  private function viewDefaultElement(FieldItemListInterface $items, EntityReferenceItem $item, Media $media, $langcode, $view_mode = "default") {
-    return [$this->entityTypeManager->getViewBuilder('media')->view($media, $view_mode)];
+    $element['#pre_render'][] = [MediaResponsiveImageFormatter::class, 'preRender'];
```

Comment:
This is cheeky. I was not aware that the pre_render function took this kind of parameter. Thanks for the refactor.